app/pages/learn/01_tutorial/04_mastering-the-api/02_invoke/00_methodhandle.md

```diff
@@ -0,0 +1,531 @@
+
+```java
+Example example = new Example();
+```
+
+Afterward, we must provide an instance of `Example` for invocation of our setter and getter:
+
+```java
+setterMethodHandle.invoke(example, "new value");
+String result = (String) getterMethodHandle.invoke(example); // result == `new value`
+```
+
+Although it is possible to read and write field values using method handles, it's not common practice. 
+For fields, it's more suitable to use `VarHandle`s instead, which can be created using `findVarHandle` and `findStaticVarHandle` methods.
+
+<a id="arrays">&nbsp;</a>
+## Working with arrays
+The `MethodHandles` class contains certain methods that provide a number of preset method handles. 
+These include method handles that allow array manipulations. Creating these method handles doesn't require access checking, so the lookup object is not necessary.
+
+Let's create an array of Strings containing 5 elements using `arrayConstructor`:
+
+```java
+MethodHandle arrayConstructor = MethodHandles.arrayConstructor(String[].class);
+String[] arr = (String[]) arrayConstructor.invoke(5);
+```
+
+To modify a single element, we can use `arrayElementSetter`, to which we provide the reference to the target array, the index of an element, and the new value:
+
+```java
+MethodHandle elementSetter = MethodHandles.arrayElementSetter(String[].class);
+elementSetter.invoke(arr, 4, "test");
+```
+
+To read the value of a single element, we should use `arrayElementGetter` method handle, to which we provide the reference to an array and the element index:
+
+```java
+MethodHandle elementGetter = MethodHandles.arrayElementGetter(String[].class);
+String element = (String) elementGetter.invoke(arr, 4); // element == "test"
+```
+
+We could also use the method handle provided by `arrayLength` to get the array length as integer:
+
+```java
+MethodHandle arrayLength = MethodHandles.arrayLength(String[].class);
+int length = (int) arrayLength.invoke(arr); // length == 5
+```
+
+<a id="vsreflection">&nbsp;</a>
+## Method handles vs Reflection
+The Method Handles mechanism was not designed for wide usage. Originally, it was meant as a tool to assist compiler and language runtime developers. 
+It has a narrower range of capabilities than the Reflection API. 
+For example, the Reflection API allows listing the class members and inspecting their properties, which cannot be done using method handles.
+
+There are also certain differences when it comes to access checking and security considerations. 
+The Core Reflection API performs access checking against every caller, on every call, while for method handles, access is checked only during construction. 
+This makes invocation through method handles faster than through reflection. 
+However, certain precautions have to be taken so the method handle is not passed to the code where it shouldn't be accessible.
+
+Method handles and reflection serve different purposes, so it's important to pick the right tool based on your needs.
```

Comment:
When saying it's important to pick the right tool, it might be a good idea to specify how to do that.